app/src/main/java/com/example/tasky/agenda/data/EventRepositoryImpl.kt

```diff
@@ -2,15 +2,34 @@ package com.example.tasky.agenda.data
 
 import com.example.tasky.agenda.data.local.AgendaDao
 import com.example.tasky.agenda.data.mapper.toDomain
+import com.example.tasky.agenda.data.remote.AgendaService
 import com.example.tasky.agenda.domain.EventRepository
 import com.example.tasky.agenda.domain.model.AgendaItem
+import com.example.tasky.agenda.domain.model.Attendee
+import com.example.tasky.core.data.Resource
+import com.example.tasky.core.data.remote.safeApiCall
 import javax.inject.Inject
 
 class EventRepositoryImpl @Inject constructor(
-    private val agendaDao: AgendaDao
+    private val agendaDao: AgendaDao,
+    private val agendaService: AgendaService
 ) : EventRepository {
 
     override suspend fun getEventById(eventId: String): AgendaItem.Event {
         return agendaDao.getEventById(eventId).toDomain()
     }
+
+    override suspend fun getValidUser(email: String): Attendee? {
+        return when (val result = safeApiCall { agendaService.getValidUser(email) }) {
+            is Resource.Success -> {
+                if (result.data.doesUserExist) {
+                    result.data.attendee?.toDomain()
+                } else {
+                    null
```

Comment:
Here you could parse the error from the server, and return it

Reply:
![image](https://github.com/MelvinKucuk/tasky/assets/42274125/9db8c6fe-f041-407f-9d92-411d46924d4e)
When the user doesnt exists the query returns 200 with doesUserExist as false